libs/sdk-core/src/breez_services.rs

```diff
@@ -2096,57 +2092,23 @@ impl Receiver for PaymentReceiver {
         info!("Existing routing hints {:?}", parsed_invoice.routing_hints);
-        info!("lsp info pubkey = {:?}", lsp_info.pubkey.clone());
-        let has_lsp_hint = parsed_invoice.routing_hints.iter().any(|h| {
-            h.hops
-                .iter()
-                .any(|h| h.src_node_id == lsp_info.pubkey.clone())
-        });
 
-        // We only add routing hint if we need to open a channel
-        // or if the invoice doesn't have any routing hints that points to the lsp
-        let mut lsp_hint: Option<RouteHint> = None;
-        if !has_lsp_hint || open_channel_needed {
-            match short_channel_id {
-                Some(short_channel_id) => {
-                    let lsp_hop = RouteHintHop {
-                        src_node_id: lsp_info.pubkey,
-                        short_channel_id,
-                        fees_base_msat: lsp_info.base_fee_msat as u32,
-                        fees_proportional_millionths: (lsp_info.fee_rate * 1000000.0) as u32,
-                        cltv_expiry_delta: lsp_info.time_lock_delta as u64,
-                        htlc_minimum_msat: Some(lsp_info.min_htlc_msat as u64),
-                        htlc_maximum_msat: None,
-                    };
-
-                    info!("Adding LSP hop as routing hint: {:?}", lsp_hop);
-                    lsp_hint = Some(RouteHint {
-                        hops: vec![lsp_hop],
-                    });
-                }
-                None => info!("No available channel ID for route hint"),
-            }
-        }
-
-        // We only create a new invoice if we need to add the lsp hint or change the amount
-        if lsp_hint.is_some() || req.amount_msat != destination_invoice_amount_msat {
-            // create the large amount invoice
-            let raw_invoice_with_hint = add_lsp_routing_hints(
-                invoice.clone(),
-                !open_channel_needed,
-                lsp_hint,
-                req.amount_msat,
-            )?;
+        // In the case we have liquidity we only append the routing hints we have to the gl invoice.
+        // In the case we don't have liquidity we set the lsp hint to the gl invoice and change the amount.
+        let raw_invoice_with_hint = add_routing_hints(
+            invoice.clone(),
+            !open_channel_needed,
+            &routing_hints,
+            req.amount_msat,
+        )?;
 
-            info!("Routing hint added");
-            let signed_invoice_with_hint = self.node_api.sign_invoice(raw_invoice_with_hint)?;
-            info!("Signed invoice with hint = {}", signed_invoice_with_hint);
+        info!("Routing hint added");
+        let signed_invoice_with_hint = self.node_api.sign_invoice(raw_invoice_with_hint)?;
+        info!("Signed invoice with hint = {}", signed_invoice_with_hint);
 
-            parsed_invoice = parse_invoice(&signed_invoice_with_hint)?;
-        }
+        parsed_invoice = parse_invoice(&signed_invoice_with_hint)?;
```

Comment:
I would suggest trying to ascertain if the routing hints have changed and only signing a new invoice if its the case. Else we're always creating two invoices, even if the route hints are no different to the GL ones.

Reply:
done